homeassistant/__main__.py

```diff
@@ -31,76 +29,34 @@ def ensure_pip():
         sys.exit()
 
 
-# Copy of homeassistant.util.package because we can't import yet
-def install_package(package):
-    """Install a package on PyPi. Accepts pip compatible package strings.
-    Return boolean if install successfull."""
-    args = [sys.executable, '-m', 'pip', 'install', '--quiet', package]
-    if not IS_VIRTUAL:
-        args.append('--user')
-    try:
-        return 0 == subprocess.call(args)
-    except subprocess.SubprocessError:
-        return False
-
-
-def validate_dependencies():
-    """ Validate all dependencies that HA uses. """
-    ensure_pip()
-
-    print("Validating dependencies...")
-    import_fail = False
-
-    for requirement in DEPENDENCIES:
-        if not install_package(requirement):
-            import_fail = True
-            print('Fatal Error: Unable to install dependency', requirement)
-
-    if import_fail:
-        print(("Install dependencies by running: "
-               "python3 -m pip install -r requirements.txt"))
-        sys.exit()
-
-
-def ensure_path_and_load_bootstrap():
-    """ Ensure sys load path is correct and load Home Assistant bootstrap. """
-    try:
-        from homeassistant import bootstrap
-
-    except ImportError:
-        # This is to add support to load Home Assistant using
-        # `python3 homeassistant` instead of `python3 -m homeassistant`
-
-        # Insert the parent directory of this file into the module search path
-        sys.path.insert(0, os.path.join(os.path.dirname(__file__), '..'))
-
-        from homeassistant import bootstrap
-
-    return bootstrap
-
-
-def validate_git_submodules():
-    """ Validate the git submodules are cloned. """
-    try:
-        # pylint: disable=no-name-in-module, unused-variable
-        from homeassistant.external.noop import WORKING  # noqa
-    except ImportError:
-        print("Repository submodules have not been initialized")
-        print("Please run: git submodule update --init --recursive")
-        sys.exit()
-
-
 def ensure_config_path(config_dir):
     """ Gets the path to the configuration file.
         Creates one if it not exists. """
 
+    lib_dir = os.path.join(config_dir, 'lib')
+
     # Test if configuration directory exists
     if not os.path.isdir(config_dir):
-        print(('Fatal Error: Unable to find specified configuration '
-               'directory {} ').format(config_dir))
-        sys.exit()
-
-    import homeassistant.config as config_util
+        if config_dir == config_util.get_default_config_dir():
+            try:
+                os.mkdir(config_dir)
+            except OSError:
+                print(('Fatal Error: Unable to create default configuration '
+                       'directory {} ').format(config_dir))
+                sys.exit()
```

Comment:
I know this was already here but let's change this to `sys.exit(1)` to show the system that an error caused Home Assistant to quit.